ctapipe/io/dl1eventsource.py

```diff
@@ -313,9 +319,13 @@ def _generate_events(self):
             data.count = counter
             data.trigger = trigger
             data.index = index
-            data.trigger.tels_with_trigger = self.subarray.tel_mask_to_tel_ids(
+            data.trigger.tels_with_trigger = self._full_subarray_info.tel_mask_to_tel_ids(
```

Comment:
If you want to be consistent, this should not be the full subarray info, but the selected subarray.  Otherwise we would have to store both in the output file in order to understand how to convert between index and id.   I think there is no real reason to keep the full subarray around. . 

For example, if we subselect only 4 telescopes, it would be correct to have the tel_mask be only 4 elements wide, rather than whatever the super-array was (which could be very large for some simulations).  The SubarrayDescription is always what is used to map indices to tel_ids.

Reply:
I think here it needs to be read with the full array (like it is in the file) and then converted to another mask later. So I think this here is correct.

Reply:
But this is an array of booleans with length equal to the number of telescopes in the full array.
I think the easiest way to handle it is to select from the full subarray and only choose the allowed telescopes afterwards, because you cant just match that mask with the reduced subarray.

If you see an easier way, I'm open to change it, I just failed to think of one

Reply:
ah, yes I understand now why this is not easy (it's a sub-selection a the DL1 level after a possible sub-selection at the DL0 level). Ok, fine then, you can ignore my reques.